lightning/src/ln/peer_handler.rs

```diff
@@ -1208,20 +1215,23 @@ mod tests {
 
 		for i in 0..peer_count {
 			let router = test_utils::TestRoutingMessageHandler::new();
+			let router = if let Some(routers) = routing_handlers { routers[i].clone() } else {
+				Arc::new(router)
+			};
```

Comment:
Would be nice to avoid the duplicate variable name here - Rust allows it, but it tends to make code hard to read (and I really, really, really wish Rust didn't allow it).

```suggestion
			let router = if let Some(routers) = routing_handlers { routers[i].clone() } else {
				Arc::new(test_utils::TestRoutingMessageHandler::new())
			};
```